### ISSUE TITLE
[BUG]“Current loss scale already at minimum - cannot decrease scale anymore” when I enabled ZERO++ 

### ISSUE THREAD

**Issue by janelu9** [microsoft/DeepSpeed#3856]
deepspeed 0.10.0
here is my config：
```
 json = {
    "train_batch_size": 2,
    "train_micro_batch_size_per_gpu": 2,
    "steps_per_print": 20,
    "zero_allow_untested_optimizer": true,
    "zero_optimization": {
        "stage": 3,
        "reduce_bucket_size": 1.000000e+07,
        "reduce_scatter": true,
        "zero_quantized_weights": true,
        "zero_hpz_partition_size": 16,
        "zero_quantized_gradients": true,
        "contiguous_gradients": true,
        "overlap_comm": true,
        "offload_param": {
            "device": "none"
        },
        "offload_optimizer": {
            "device": "none"
        },
        "stage3_param_persistence_threshold": 1.000000e+04,
        "stage3_max_live_parameters": 3.000000e+07,
        "stage3_prefetch_bucket_size": 3.000000e+07,
        "memory_efficient_linear": false
    },
    "fp16": {
        "enabled": true,
        "loss_scale_window": 100
    },
    "gradient_clipping": 1.0,
    "prescale_gradients": false,
    "wall_clock_breakdown": false,
    "hybrid_engine": {
        "enabled": false,
        "max_out_tokens": 2.048000e+03,
        "inference_tp_size": 1,
        "release_inference_cache": false,
        "pin_parameters": true,
        "tp_gather_partition_size": 8
    }
}
```
model and params:
```
model=LlamaForCausalLM.from_pretrained("decapoda-research/llama-13b-hf")
args.zero_stage=3
args.num_train_epochs=1
args.per_device_train_batch_size = 2
args.gradient_accumulation_steps =1
args.seed=1234
args.weight_decay=0.01
args.lr_scheduler_type="cosine"
args.num_warmup_steps=500
args.learning_rate=1e-5
args.output_dir = "."
```

**Comment by starlitsky2010** [comment]
I also met this peoblem and my deepspeed version is  0.8.3.
Here is my zero2_config.json

{
  "fp16": {
      "enabled": "auto",
      "loss_scale": 0,
      "loss_scale_window": 1000,
      "initial_scale_power": 16,
      "hysteresis": 2,
      "min_loss_scale": 1
  },

  "bf16": {
      "enabled": "auto"
  },

  "optimizer": {
      "type": "AdamW",
      "params": {
          "lr": "auto",
          "betas": "auto",
          "eps": "auto",
          "weight_decay": "auto"
      }
  },

  "scheduler": {
      "type": "WarmupLR",
      "params": {
          "warmup_min_lr": "auto",
          "warmup_max_lr": "auto",
          "warmup_num_steps": "auto"
      }
  },

  "zero_optimization": {
      "stage": 2,
      "offload_optimizer": {
          "device": "cpu",
          "pin_memory": true
      },
      "allgather_partitions": true,
      "allgather_bucket_size": 2e8,
      "overlap_comm": true,
      "reduce_scatter": true,
      "reduce_bucket_size": "auto",
      "contiguous_gradients": true
  },

  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "steps_per_print": 2000,
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto",
  "wall_clock_breakdown": false
}

**Comment by starlitsky2010** [comment]
What aspect we should focus when met this problem? Any suggestions?

**Comment by HeyangQin** [comment]
Hello @starlitsky2010. For ZeRO stage 2, this is likely a different issue. You can try to update DeepSpeed to the latest version. If the problem is still there, please create a new issue.
